pkg/downloader/chart_downloader.go

```diff
@@ -103,10 +113,17 @@ func (c *ChartDownloader) DownloadTo(ref, version, dest string) (string, *proven
 	}
 
 	name := filepath.Base(u.Path)
-	if u.Scheme == registry.OCIScheme {
+	if u.Scheme == registry.OCIScheme || u.Scheme == "git" {
```

Comment:
This new condition doesn't seem to do anything, because if it's true, then `scheme` would also be `"git"`, and so the next block will immediately overwrite `name` anyway. I also don't see how this block would apply to git sources anyway, they don't separate the chart name and version with a `colon` like OCI sources.